surveys/views_api.py

```diff
@@ -23,3 +24,17 @@ def get_serializer(self, *args, **kwargs):
         kwargs['data']['survey'] = self.survey.pk
         serializer = super().get_serializer(*args, **kwargs)
         return serializer
+
+
+class SurveyLatestView(generics.ListAPIView):
+    queryset = UserResponse.objects.all()
+
+    def dispatch(self, request, *args, **kwargs):
+        survey_pk = kwargs.pop('pk')
+        self.user_id = kwargs.pop('user_id')
+        self.survey = get_object_or_404(Survey.objects.all(), pk=survey_pk)
+        return super().dispatch(request, *args, **kwargs)
+
+    def list(self, request, *args, **kwargs):
+        data = self.get_queryset().latest_for_user(self.survey, self.user_id)
```

Comment:
As `self.user_id` is coming for the url, it can contain a non existing `user_id` which would create an IndexError.